### PR TITLE
MULE-19793: Replace semantic terms enricher with a proper parser

### PR DESCRIPTION
**Pull request by marianogonzalez**

### CODE REVIEW

File: modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/java/ParameterModelsLoaderDelegate.java [hunk 1]
```diff
@@ -44,39 +44,41 @@
...
-        extensionParameter.getDeprecationModel().ifPresent(parameter::withDeprecation);
-        extensionParameter.getDisplayModel().ifPresent(parameter::withDisplayModel);
-        extensionParameter.getAdditionalModelProperties().forEach(parameter::withModelProperty);
+        parameterParser.getLayoutModel().ifPresent(parameter::withLayout);
+        parameterParser.getDslConfiguration().ifPresent(parameter::withDsl);
+        parameterParser.getDeprecationModel().ifPresent(parameter::withDeprecation);
+        parameterParser.getDisplayModel().ifPresent(parameter::withDisplayModel);
+        parameterParser.getAdditionalModelProperties().forEach(parameter::withModelProperty);
+        addSemanticTerms(parameter.getDeclaration(), parameterParser);
+        parameter.getDeclaration().getSemanticTerms().addAll(parameterParser.getSemanticTerms());
```
Comment (ndinu) [inline modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/java/ParameterModelsLoaderDelegate.java:81]: these two lines seems confusing, aren't they doing the same?

File: modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/java/RouterModelLoaderDelegate.java [hunk 2]
```diff
@@ -47,7 +49,7 @@ void declareRouter(ExtensionDeclarer extensionDeclarer, HasConstructDeclarer own
     parser.getAdditionalModelProperties().forEach(router::withModelProperty);
 
     loader.getParameterModelsLoaderDelegate().declare(router, parser.getParameterGroupModelParsers());
-
+    addSemanticTerms(router.getDeclaration(), parser);
```
Comment (ndinu) [inline modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/java/RouterModelLoaderDelegate.java:52]: does this also needs to be added to its nested routes?

File: modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/java/ParameterModelsLoaderDelegate.java (hunk 1 above)
Comment (marianogonzalez) [reply modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/java/ParameterModelsLoaderDelegate.java:81]: good catch. Fixed

File: modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/java/RouterModelLoaderDelegate.java (hunk 2 above)
Comment (marianogonzalez) [reply modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/java/RouterModelLoaderDelegate.java:52]: no

**Review by ndinu — APPROVED** [review]

**Pull request merged**